core/src/main/scala/com/yahoo/maha/core/query/oracle/OracleOuterGroupByQueryGenerator.scala

```diff
@@ -488,22 +488,28 @@ abstract class OuterGroupByQueryGenerator(partitionColumnRenderer:PartitionColum
               } else {
                 s""""$alias"""
               }
-            case FactColumnInfo(alias) =>
-              val column  = if(queryBuilderContext.containsFactAliasToColumnMap(alias)) {
-                queryBuilderContext.getFactColByAlias(alias)
-              } else {
-                // Case to handle CustomRollup Columns
-                val aliasToColNameMap: Map[String, String] = factBest.factColMapping.map {
-                  case (factColName, factAlias) =>
-                    val col = factBest.fact.columnsByNameMap(factColName)
-                    val name:String = col.alias.getOrElse(col.name)
-                    factAlias -> name
-                }
-                require(aliasToColNameMap.contains(alias), s"Can not find the alias $alias in aliasToColNameMap")
-                val colName = aliasToColNameMap(alias)
-                queryBuilderContext.getFactColByAlias(colName)
-              }
+            case FactColumnInfo(alias) if queryBuilderContext.containsFactAliasToColumnMap(alias) =>
+              val column = queryBuilderContext.getFactColByAlias(alias)
               renderParentOuterDerivedFactCols(alias, column)
+              /*TODO: I am unable to generate a test case for the else condition below with any combinations of
```

Comment:
@pranavbhole I couldn't create a test case with any combination of custom rollup and derived column to get the test to get into this if statement.  If you have an example, please add to this branch, I gave you access to my repo so you can push to this branch (github.com/patelh/maha).